webapp/src/containers/Rhs/projectList/index.tsx

```diff
@@ -1,47 +1,76 @@
 import {useDispatch} from 'react-redux';
 
 import ProjectCard from 'components/card/project';
+import EmptyState from 'components/emptyState';
+import LinearLoader from 'components/loader/linear';
+import ConfirmationModal from 'components/modal/confirmationModal';
 
 import {setProjectDetails} from 'reducers/projectDetails';
-
-// TODO: dummy data, remove later
-const data: ProjectDetails[] = [
-    {
-        id: 'abc',
-        title: 'Project A',
-        organization: 'Organization Name',
-    },
-    {
-        id: 'abc1',
-        title: 'Project B',
-        organization: 'Organization Name',
-    },
-    {
-        id: 'abc2',
-        title: 'Project C',
-        organization: 'Organization Name',
-    },
-];
+import {showLinkModal} from 'reducers/linkModal';
+import usePluginApi from 'hooks/usePluginApi';
+import plugin_constants from 'plugin_constants';
 
 const ProjectList = () => {
+    const [showConfirmationModal, setShowConfirmationModal] = useState(false);
     const dispatch = useDispatch();
+    const usePlugin = usePluginApi();
 
     const handleProjectTitleClick = (projectDetails: ProjectDetails) => {
         dispatch(setProjectDetails(projectDetails));
     };
 
+    const handleOpenLinkProjectModal = () => {
+        dispatch(showLinkModal([]));
+    };
+
+    useEffect(() => {
+        usePlugin.makeApiRequest(plugin_constants.pluginApiServiceConfigs.getAllLinkedProjectsList.apiServiceName);
+    }, []);
+
+    const data = usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.getAllLinkedProjectsList.apiServiceName).data;
+
     return (
         <>
             <p className='rhs-title'>{'Linked Projects'}</p>
             {
-                data.map((item) => (
-                    <ProjectCard
-                        onProjectTitleClick={handleProjectTitleClick}
-                        projectDetails={{id: item.id, title: item.title, organization: item.organization}}
-                        key={item.id}
-                    />
-                ),
+                <ConfirmationModal
+                    isOpen={showConfirmationModal}
+                    onHide={() => setShowConfirmationModal(false)}
+                    onConfirm={() => setShowConfirmationModal(false)}
+                    confirmBtnText='Unlink'
+                    description='Are you sure you want to unlink this Project?'
+                    title='Confirm Project Unlink'
+                />
+            }
+            {
+                usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.getAllLinkedProjectsList.apiServiceName).isLoading && (
+                    <LinearLoader/>
+                )
+            }
+            {
+                usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.getAllLinkedProjectsList.apiServiceName).isSuccess &&
+                data &&
+                (
+                    data?.length > 0 ?
+                        data?.map((item) => (
```

Comment:
```suggestion
                        data.map((item) => (
```
You've already added a check for `data` above.